packages-exp/auth-exp/src/core/user/token_manager.ts

```diff
@@ -61,6 +62,38 @@ export class StsTokenManager {
     throw new Error('StsTokenManager: token refresh not implemented');
   }
 
+  toPlainObject(): object {
+    return {
+      refreshToken: this.refreshToken,
+      accessToken: this.accessToken,
+      expirationTime: this.expirationTime
+    };
+  }
+
+  static fromPlainObject(
+    appName: string,
+    object: { [key: string]: unknown }
+  ): StsTokenManager {
+    const { refreshToken, accessToken, expirationTime } = object;
+    const internalError = AUTH_ERROR_FACTORY.create(
+      AuthErrorCode.INTERNAL_ERROR,
+      { appName }
+    );
+    if (refreshToken && typeof refreshToken !== 'string') {
+      throw internalError;
+    }
+    if (accessToken && typeof accessToken !== 'string') {
+      throw internalError;
+    }
+    if (expirationTime && typeof expirationTime !== 'number') {
+      throw internalError;
```

Comment:
should be assertions instead?

Reply:
The assertion in @firebase/util comes out as a database error which seems wrong

Reply:
https://github.com/firebase/firebase-js-sdk/blob/master/packages/util/src/assert.ts

Reply:
oh weird, maybe we should add an authAssert?

Reply:
Done